### PR TITLE
Fix out-of-bound in the initialization of EB

### PR DESCRIPTION
**Pull request by RemiLehe**
When compiling with EB in debug mode, I got an out-of-bound message at this line:
https://github.com/ECP-WarpX/WarpX/blob/development/Source/EmbeddedBoundary/WarpXInitEB.cpp#L140
It seems that some of the internal structures for EB in `amrex` (used e.g. in `getType`) have only 1 guard cells, whereas we pass a box that has more guard cells.

~~Instead, this PR does not add the guard cells in the box passed in `getType`, and instead the guard cells are filled by an additional `FillBoundary`.~~
Instead, this PR calls `getType` by passing a box that does not have guard cells (but adds the guard cells when actually touching the `edge_length` and `face_area` arrays.

### CODE REVIEW

**Comment by lgiacome** [comment]
I don't think I fully understand why this happens, but your solution looks great to me. We need the same changes in `ScaleEdges `, `ScaleAreas` and `MarkCells`, don't we?

**Comment by RemiLehe** [comment]
I don't think we need the same change for `ScaleEdges`, `ScaleAreas` and `MarkCells`, because these do not call the function `getType`. 

The problem is really with `getType`, since it seems that, under the hood, `getType` looks up an internal EB structure which has only *one* guard cell, and does so for every single cell of the `box` that is passed to `getType`.

**Comment by lgiacome** [comment]
Now I see, thanks a lot for the explanation.
Unfortunately, all the CI tests with EM and EB fail and it seems like the new changes don't work, but on my laptop they compile fine :/ is there a way to see a full traceback from the azure tests?

**Comment by EZoni** [comment]
> Unfortunately, all the CI tests with EM and EB fail and it seems like the new changes don't work, but on my laptop they compile fine :/ is there a way to see a full traceback from the azure tests?

@lgiacome Hi Lorenzo, in the table with all the checks on this page here you can click the link "Details" next to the red ones (the ones that failed), then click the link that will say "1 errors / 0 warnings" (or something similar), then click on the failing job (e.g. "embedded boundary" in this case), then click on "Build & test" and you'll see on your right the raw log with the backtraces printed out for the individual tests that failed (you can also view/download this raw log in a separate window, if needed). Keep in mind that those backtraces are not generated in DEBUG mode (we don't run in DEBUG mode on Azure), so the information might not be as clear and useful as the one you get in DEBUG mode. Anyways let me know if you find those backtraces by following these steps!

**Update** Note that I do see some tests crashing even locally. For instance, if I run locally `./run_test.sh embedded_boundary_cube` on this branch `fix_eb_out_of_bound`, the test does crash, with backtraces. Hope this is useful info for debugging.

**Comment by lgiacome** [comment]
@EZoni thanks a lot for both the points! I'll look into running this locally then and see what's going on.

File: Source/EmbeddedBoundary/WarpXInitEB.cpp [hunk 1]
```diff
@@ -177,6 +177,14 @@ WarpX::ComputeEdgeLengths () {
...
         }
     }
+    // Echange guard cells
+    for (int idim = 0; idim < 3; ++idim){
+#ifdef WARPX_DIM_XZ
+        if(idim == 1) continue;
+#endif
+        m_edge_lengths[maxLevel()][idim]->FillBoundary();
+    }
+
```
Comment (RemiLehe) [inline Source/EmbeddedBoundary/WarpXInitEB.cpp]:
```suggestion
```

File: Source/EmbeddedBoundary/WarpXInitEB.cpp [hunk 2]
```diff
@@ -238,6 +246,14 @@ WarpX::ComputeFaceAreas () {
...
         }
     }
+    // Echange guard cells
+    for (int idim = 0; idim < 3; ++idim){
+#ifdef WARPX_DIM_XZ
+        if(idim == 1) continue;
+#endif
+        m_face_areas[maxLevel()][idim]->FillBoundary();
+    }
+
```
Comment (RemiLehe) [inline Source/EmbeddedBoundary/WarpXInitEB.cpp]:
```suggestion
```

**Review by lgiacome — APPROVED** [review]
Looks great thanks Remi 👍

**Review by ax3l — APPROVED** [review]

**Pull request merged**